src/lifetracker/logic/CommandLineResult.java

```diff
@@ -35,4 +40,20 @@ public void addResultLine(String resultLine) {
 
         resultLines.add(resultLine);
     }
+    
+    @Override
+    public void setType(String commandString){
+    	if(commandString.equals("exit")){
+    		this.commandType = CommandType.EXIT;
+    		System.exit(0);
```

Comment:
Do not exit here! 

This is a premature termination. Just pass the signal to the UI and let it handle the exit. If you simply exit here, the storage will shutdown immediately and cause file corruptions.